src/compiler/transformers/add-static-style.ts

```diff
@@ -134,16 +134,67 @@ const createStyleLiteral = (cmp: d.ComponentCompilerMeta, style: d.StyleCompiler
   return ts.factory.createStringLiteral(style.styleStr);
 };
 
-const createStyleIdentifierFromUrl = (cmp: d.ComponentCompilerMeta, style: d.StyleCompiler) => {
-  style.styleIdentifier = dashToPascalCase(cmp.tagName);
-  style.styleIdentifier = style.styleIdentifier.charAt(0).toLowerCase() + style.styleIdentifier.substring(1);
-
+export const createStyleIdentifier = (cmp: d.ComponentCompilerMeta, style: d.StyleCompiler) => {
+  /**
+   * remove duplicates in `styleUrl`
+   */
+  const externalStyles = Array.from(new Set(style.externalStyles.map((s) => s.absolutePath)));
```

Comment:
Excellent observation regarding the import order! I'll ensure to account for this.

Reply:
Changes applied:
```ts
/**
   * remove duplicates in `styleUrl`
   */
  const externalStyles = style.externalStyles.map((s) => s.absolutePath)
    /**
     * Reverse order and reduce to remove duplicates. This will make sure that duplicate
     * styles applied to the same component will be applied in the order they are
     * defined in the component, e.g.
     * ```
     * @Component({
     *  styleUrls: ['cmp-a.css', 'cmp-b.css', 'cmp-a.css']
     * })
     * ```
     * will be applied in the order `cmp-b.css`, `cmp-a.css`.
     */
    .reverse()
    .reduce((extStyles, styleUrl) => {
      if (!extStyles.includes(styleUrl)) {
        extStyles.push(styleUrl);
      }
      return extStyles;
    }, [])
    /**
     * Reverse back to the original order
     */
    .reverse();
```